common/config_test.go

```diff
@@ -55,86 +76,135 @@ func TestDownloadableURL(t *testing.T) {
+				"test-fixtures\\SomeDir\\myfile.txt",
+				fmt.Sprintf("file:///%s/SomeDir/myfile.txt", portablepath),
+				false,
 			},
-			{ // need windows drive
-				"/Temp/SomeDir/myfile.txt",
-				"",
-				true,
+			{ // without the drive makes this native path a relative file:// uri
+				"test-fixtures/SomeDir/myfile.txt",
+				fmt.Sprintf("file:///%s/SomeDir/myfile.txt", portablepath),
+				false,
 			},
-			{ // UNC paths; why not?
-				"\\\\?\\c:\\Temp\\SomeDir\\myfile.txt",
-				"",
-				true,
+			{ // UNC paths being promoted to smb:// uri scheme.
+				fmt.Sprintf("\\\\localhost\\C$\\%s\\SomeDir\\myfile.txt", nativepath),
+				fmt.Sprintf("smb://localhost/C$/%s/SomeDir/myfile.txt", portablepath),
+				false,
 			},
-			{
-				"file:///C:\\Temp\\SomeDir\\myfile.txt",
-				"file:///c:/Temp/SomeDir/myfile.txt",
+			{ // Absolute uri (incorrect slash type)
+				fmt.Sprintf("file:///%s\\SomeDir\\myfile.txt", nativepath),
+				fmt.Sprintf("file:///%s/SomeDir/myfile.txt", portablepath),
 				false,
 			},
-			{
-				"file:///c:/Temp/Somedir/myfile.txt",
-				"file:///c:/Temp/SomeDir/myfile.txt",
+			{ // Absolute uri (existing and mis-spelled)
+				fmt.Sprintf("file:///%s/Somedir/myfile.txt", nativepath),
+				fmt.Sprintf("file:///%s/SomeDir/myfile.txt", portablepath),
+				false,
+			},
+			{ // Absolute path (non-existing)
+				"\\absolute\\path\\to\\non-existing\\file.txt",
+				"file:///absolute/path/to/non-existing/file.txt",
+				false,
+			},
+			{ // Absolute paths (existing)
+				fmt.Sprintf("%s/SomeDir/myfile.txt", nativepath),
+				fmt.Sprintf("file:///%s/SomeDir/myfile.txt", portablepath),
+				false,
+			},
+			{ // Relative path (non-existing)
+				"./nonexisting/relative/path/to/file.txt",
+				"file://./nonexisting/relative/path/to/file.txt",
+				false,
+			},
+			{ // Relative path (existing)
+				"./test-fixtures/SomeDir/myfile.txt",
+				fmt.Sprintf("file:///%s/SomeDir/myfile.txt", portablepath),
+				false,
+			},
+			{ // Absolute uri (existing and with `/` prefix)
+				fmt.Sprintf("file:///%s/SomeDir/myfile.txt", portablepath),
+				fmt.Sprintf("file:///%s/SomeDir/myfile.txt", portablepath),
+				false,
+			},
+			{ // Absolute uri (non-existing and with `/` prefix)
+				"file:///path/to/non-existing/file.txt",
+				"file:///path/to/non-existing/file.txt",
+				false,
+			},
+			{ // Absolute uri (non-existing and missing `/` prefix)
+				"file://path/to/non-existing/file.txt",
+				"file://path/to/non-existing/file.txt",
+				false,
+			},
+			{ // Absolute uri and volume (non-existing and with `/` prefix)
+				"file:///T:/path/to/non-existing/file.txt",
+				"file:///T:/path/to/non-existing/file.txt",
+				false,
+			},
+			{ // Absolute uri and volume (non-existing and missing `/` prefix)
+				"file://T:/path/to/non-existing/file.txt",
+				"file://T:/path/to/non-existing/file.txt",
 				false,
 			},
 		}
-		// create absolute-pathed tempfile to play with
```

Comment:
I like the work you did to broaden these tests, but I'd actually still like to keep some of these hardcoded "if we're on windows, run these tests" paths with the absolute pathing.  mwhooker and I, the two Hashicorp-employed Packer maintainers, both use OSX machines, and having the gut-check of a windows-specific path test running in our Travis builds will make me feel just a little better about pathing moving foward.

Reply:
Ok. Yea, so the reason I removed the explicit windows check is because they actually should work cross-platform. `GetPortablePathToTestFixtures()` returns a proper path regardless of platform (due to `filepath`) and since `FileExistsLocally` is intended to be cross-platform, I figured it'd be good to test across all supported platforms.

But..

Reply:
You still do have an explicit check though before running these tests, in line 113, which you'll want to remove if you want this test run on multiple platforms.  I thought it over though, and you're right. Your modification to this test and the FileExistsLocally test are good.  I'm being paranoid.